external/mmdetection/detection_tasks/apis/detection/inference_task.py

```diff
@@ -218,15 +218,17 @@ def _add_predictions_to_dataset(self, prediction_results, dataset, confidence_th
                             polygon = Polygon(points=points)
                             if polygon.get_area() > 1e-12:
                                 shapes.append(Annotation(polygon, labels=labels, id=ID(f"{label_idx:08}")))
+                    if add_saliency_map and features is not None:
+                        features.append(draw_instance_segm_saliency_map(shapes, dataset_item, self._labels))
```

Comment:
IMO, you can move this to `add_features_to_data_item` to simplify the code here.

Reply:
Great suggestion! It has really simplified the `add_features_to_data_item` function call. Thank you